### PR TITLE
Additional error handling edge cases

### PR DESCRIPTION
**Pull request by obligaron**
### 🤔 What's changed?

- Added additional error handling edge cases as bad test data
- [c] Optimise error handling for empty datatable rows
- [Perl] Optimise error handling for unclosed DocStrings

### ⚡️ What's your motivation? 

These error handling edge cases were reported to the .NET implementation and fixed in an upstream project in 2019. Later, these fixes were also merged into the gherkin repository.
While working on some optimisations (see #344), the question arose whether the regression tests should be ported to the gherkin repository. This ensures that all implementations can handle these edge cases that have been reported by users.

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, etc. without changing behaviour)
- :bug: Bug fix (non-breaking change which fixes a defect)

### ♻️ Anything particular you want feedback on?

The Perl implementation in general (first time writing Perl 😉), and also whether the required check for EOF should be added to other cases in the Perl implementation (but maybe this is something for another PR).

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://github.com/cucumber/.github/tree/main?tab=coc-ov-file)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: perl/lib/Gherkin/TokenMatcher.pm [hunk 1]
```diff
@@ -238,6 +238,9 @@ sub match_StepLine {
 
 sub match_DocStringSeparator {
     my ( $self, $token ) = @_;
+    if ($token->is_eof) {
+        return;
```
Comment (mpkorstanje) [inline perl/lib/Gherkin/TokenMatcher.pm:242]: Other implementations return `true`/`false` in `_match_DocStringSeparator`, which I think should cover this case.

To make it easier for everyone to compare across languages, it would be good to be consistent.

**Review by mpkorstanje — CHANGES_REQUESTED** [review]
Looks good! 

Small request for Perl. Do you think you can do that? If not, I can take a stab at it.

File: perl/lib/Gherkin/TokenMatcher.pm (hunk 1 above)
Comment (obligaron) [reply perl/lib/Gherkin/TokenMatcher.pm]: I added an explicit false/0

File: testdata/bad/backslash_at_end_of_line_in_DataTable.feature [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Feature: Addition
+Scenario: Add two numbers
+	When I press
+		| foo |
+		| bar \
```
Comment (luke-hill) [inline testdata/bad/backslash_at_end_of_line_in_DataTable.feature:5]: Think this file should be snake cased and there is a newline missing here

Comment (obligaron) [reply testdata/bad/backslash_at_end_of_line_in_DataTable.feature]: I have changed all the testdata file names.

Regarding the missing newline: I think this is part of the tests because they should resemble someone editing the feature files in an incomplete state. The gherkin parser should parse them. So I think this is OK in this case?

I copied the tests from the [reqnroll ide plugin](https://github.com/reqnroll/Reqnroll.VisualStudio/blob/main/Tests/Reqnroll.VisualStudio.Tests/Editor/Services/DeveroomGherkinParserTests.cs).
